--- Portfile.orig	2008-10-06 17:14:34.000000000 -0700
+++ Portfile	2008-10-07 07:57:22.000000000 -0700
@@ -1,9 +1,12 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 # $Id: Portfile 38752 2008-07-30 07:16:27Z ryandesign@macports.org $
 
 PortSystem 1.0
 
 name		glade
 version		2.12.2
+set branch      [join [lrange [split ${version} .] 0 1] .]
+revision        1
 description	Glade is a free user interface builder for GTK+ and GNOME.
 long_description	Glade is a free user interface builder for GTK+ \
 			and GNOME. It is released under the GNU General \
@@ -12,24 +15,30 @@
 categories	devel gnome
 platforms	darwin
 homepage	http://glade.gnome.org/
-master_sites	gnome:sources/glade/2.12/
+master_sites	gnome:sources/${name}/${branch}/
 use_bzip2	yes
 checksums	sha1 a43d89b147296244ba4b2efc67e8772d5155afff
-	
-depends_lib \
-	bin:automake-1.7:automake bin:autoconf:autoconf \
-	lib:libiconv.2:libiconv lib:libgail-gnome:libgail-gnome \
-	lib:libXft.2.1.2:Xft2 port:freetype lib:libaudiofile:audiofile \
-	lib:libesd:esound port:fontconfig port:gnome-keyring lib:libpng:libpng \
-        lib:libatk-1.0:atk bin:gconftool-2:gconf lib:libgettextlib:gettext \
-        path:${prefix}/lib/pkgconfig/glib-2.0.pc:glib2 lib:libgnomevfs-2:gnome-vfs lib:libgtk-x11-2.0:gtk2 \
-        lib:libgnomecanvas-2:libgnomecanvas lib:libart_lgpl_2:libart_lgpl \
-        lib:libbonobo-2:libbonobo lib:libbonoboui-2:libbonoboui \
-        lib:libgnome-2:libgnome lib:libgnomeui-2:libgnomeui \
-        bin:orbit-idl-2:orbit2 port:libxml2 bin:perl:perl5.8 port:tiff \
-        path:${prefix}/lib/pkgconfig/pango.pc:pango bin:pkg-config:pkgconfig port:rarian \
-	port:xrender port:zlib
 
+patchfiles    \
+        patch-glade-gbwidgets-gbclist.c.diff \
+        patch-glade-glade_menu_editor.c.diff \
+        patch-glade-glade_keys_dialog.c.diff \
+        patch-glade-glade_project_view.h.diff
+
+depends_build \
+        port:pkgconfig \
+        port:gettext \
+        port:rarian \
+        port:gconf
+        	
+depends_lib \
+        port:gtk2 \
+        port:libxml2 \
+        port:libgnomecanvas \
+        port:libgnomeui \
+        port:libbonoboui \
+        port:libgail-gnome
+ 
 configure.cppflags-append "-L${prefix}/lib"
 
 configure.args	--disable-gnome-db
@@ -42,6 +51,9 @@
 
 variant gnomedb description {Enable GNOME DB support} {
 	configure.args-delete --disable-gnome-db
-	depends_lib-append lib:libgnomedb:libgnomedb
+	depends_lib-append port:libgnomedb
 }
 
+livecheck.check regex
+livecheck.url   http://ftp.gnome.org/pub/gnome/sources/${name}/${branch}/
+livecheck.regex {LATEST-IS-(\d+(?:\.\d+)*)}
